### DISCUSSION TITLE
"Unable to locate Avalonia.Designer.HostApp.dll for version *.*.*"

### DISCUSSION THREAD

**Discussion by BinToss** [kuiperzone/AvantGarde#25] [General]
This took me a few days to figure out.
The issue isn't that you're missing a DLL or that Avant Garde can't find one. Rather, the message indicates you're using **a build of Avant Garde which does not support the version of Avalonia UI used in your project** i.e. Avant Garde 1.3.0 supports only Avalonia 11.0.0-preview8. Later and earlier versions of Avalonia are not supported regardless of differences' insignificance.

----

I've been running into this issue with a lot of Avalonia-focused software and IDE extensions lately—they are made to support a specific version of Avalonia or a specific flavor of XAML. Correct me if I'm wrong; the project, build output, and XAML markup can provide all the necessary data for previews, or at least URIs to fetch that data. On that note, I'm very grateful for Avant Garde's "Export Avalonia Schema" feature. Despite Avant Garde being mostly useless for Avalonia UI 11.0.0-rc1.1, that specific feature has been a lifesaver in the hell that is developing in VSCode wherein developers have no acceptable previewer, let alone a designer. The working previewer we have is specifically for NoesisGui's flavor of XAML.

**Comment by kuiperzone** [comment]
Hi BinToss,

I'm not clear on this.

In order to generate a preview, Avant Garde looks at your .csproj file in order to discover what Avalonia version your project references. It then locates the "Avalonia.Designer.HostApp.dll" for that version and runs it. This host is what generates previews in terms of bitmaps, which are delivered to Avant Garde over a socket.

>"Unable to locate Avalonia.Designer.HostApp.dll for version *.*.*"

I'm not able to repeat this problem (see the screenshot below). But I am only running against 11 preview 8. I have not yet tried it against 11RC1.


> I'm very grateful for Avant Garde's "Export Avalonia Schema" feature.

I'm glad you like this. It only exists because my first attempt at building a previewer generated the previews itself, without using the Designer Host App. To do this, I had to parse XAML myself and the export feature is a hang-over from that.




![Unsaved Image 2](https://github.com/kuiperzone/AvantGarde/assets/17156238/34505a5b-5a24-41b0-b1c9-340717393c0f)

**Comment by BinToss** [comment]
> In order to generate a preview, Avant Garde looks at your .csproj file in order to discover what Avalonia version your project references. It then locates the "Avalonia.Designer.HostApp.dll" for that version and runs it. This host is what generates previews in terms of bitmaps, which are delivered to Avant Garde over a socket.

It seems this issue only occurs when running the AppImage via WSL2 Ubuntu 20.04.6 LTS (I've been putting off updating for a while now). When running Avant Garde A11 via `` or running the 1.2.1 Windows x64 release, this issue either does not occur or is not written to the preview host. Instead, the preview pane just displays "None". 
This may be because my application is a console app and will not open any windows unless started with the arguments `--config` or `--positions`. I'll continue investigating my project and try to commit all of my local changes tonight.

**Comment by kuiperzone** [comment]
Thanks for the update

**Comment by BinToss** [comment]
Got it working. I had forgotten BuildAvaloniaApp isn't allowed to be renamed. I was only able to deduce this from the new Output pane which is not present in the last stable release of Avant Garde.
![image](https://github.com/kuiperzone/AvantGarde/assets/7243190/365f7b04-6aa3-49c4-be8c-6b0387d000d9)

If VSCode or Avalonia.Diagnostics complained about it, I'd have found the issue sooner. Nothing but Avant Garde realized anything was wrong with the project.
